### PR TITLE
refac: Added tariff schema check on exceute calulation

### PR DESCRIPTION
**Pull request by energinet-ajw**
When calculating tariff charges the tariff dataframe schema is validated (asserted) against the tariff schema. 
The check is the first thing that happens in the execute method.

https://app.zenhub.com/workspaces/mandalorian-5fd22a1a59e4740018f5ab5a/issues/gh/energinet-datahub/green-energy-hub/476

### CODE REVIEW

File: source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+from package.calculation_output import WholesaleCalculationResultWriter
+
+
+@patch.object(calculation_output, WholesaleCalculationResultWriter.__name__)
+def test__execute__asserts_tariff_schema__is_valid(
+    wholesale_calculation_result_writer_mock: WholesaleCalculationResultWriter, spark
+):
+    # Arrange
+    period_start_datetime: datetime = datetime(2020, 1, 1, 0, 0)
+    tariffs_hourly_df: DataFrame = spark.createDataFrame(
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py:33]:
```suggestion
    period_start_datetime = datetime(2020, 1, 1, 0, 0)
    tariffs_hourly_df = spark.createDataFrame(
```

File: source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    # Act & Assert
+    try:
+        execute(
+            wholesale_calculation_result_writer_mock,
+            tariffs_hourly_df,
+            period_start_datetime,
+        )
+    except ValueError:
+        raise Exception("Test failed. Exception raised!")
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py:45]: Can we rewrite to [the pytest way](https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions) of doing this?
And then you'll also avoid this annotation that tells you that you have test code that is actually not executed:
image

Comment (BjarkeMeier) [reply source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py:45]: Second thought:
```suggestion
    # Act and assert (that no error is raised)
    execute(
        wholesale_calculation_result_writer_mock,
        tariffs_hourly_df,
        period_start_datetime,
    )
```

File: source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+from calculation.wholesale.test_tariff_calculators import _create_tariff_hour_row
+from package import calculation_output
+from package.calculation.wholesale import execute
+from package.calculation.wholesale.schemas.tariffs_schema import tariff_schema
+from package.calculation_output import WholesaleCalculationResultWriter
+
+
+@patch.object(calculation_output, WholesaleCalculationResultWriter.__name__)
+def test__execute__asserts_tariff_schema__is_valid(
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py:28]:
```suggestion
def test__execute__when_tariff_schema_is_valid__does_not_raise(
```

File: source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            wholesale_calculation_result_writer_mock,
+            tariffs_hourly_df,
+            period_start_datetime,
+        )
+    except ValueError:
+        raise Exception("Test failed. Exception raised!")
+
+
+@patch.object(calculation_output, WholesaleCalculationResultWriter.__name__)
+def test__execute__asserts_tariff_schema__throws_exception(
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py:49]:
```suggestion
def test__execute__when_tariff_schema_is_invalid__raises_assertion_error(
```

File: source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py (hunk 2 above)
Comment (energinet-ajw) [reply source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py]: Changed to a comment in the test instead.

File: source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    wholesale_calculation_result_writer_mock: WholesaleCalculationResultWriter, spark
+):
+    # Arrange
+    data = [("John", "Dow")]
+    period_start_datetime: datetime = datetime(2020, 1, 1, 0, 0)
+    tariffs_hourly_df: DataFrame = spark.createDataFrame(data)
+
+    # Act & Assert
+    with pytest.raises(Exception):
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/calculation/wholesale/test_wholesale_calculation.py:58]:
```suggestion
    with pytest.raises(AssertionError):
```

**Review by johevemi — APPROVED** [review]

**Pull request merged**